### PR TITLE
chore: migrate to `pnpm`

### LINKED ISSUE #249: Migrate from `yarn` to `pnpm`
## 📚 Context

We've been using `yarn` to manage JavaScript/TypeScript dependencies since the beginning.
There are other Node package managers available. One of those is [`pnpm`](https://pnpm.io/).
There are many reasons to migrate from `yarn` to `pnpm`. See [motivations](https://pnpm.io/motivation).

## ✔️ Solution

Migrate to `pnpm`

## 📈 Subtasks

- [ ] Delete the `yarn.lock` file
- [ ] Run `pnpm i` to create a `pnpm-lock.yaml` file
- [ ] Fix GitHub workflow files to use `pnpm`
- [ ] Fix commands in the README

### PR DESCRIPTION
**Pull request by guidanoli**
Resources:
- https://dev.to/andreychernykh/yarn-npm-to-pnpm-migration-guide-2n04
- https://pnpm.io/continuous-integration

Closes #249

### CODE REVIEW

**Comment by guidanoli** [comment]
The CI is failing due to an issue in the tests (recent foundry nightly release broke it :-).
The issue was addressed by #248, so let's wait until that is merged.

**Review by ZzzzHui — APPROVED** [review]

**Review by pedroargento — APPROVED** [review]

**Pull request merged**
